SwiftIntro.playground/Contents.swift

```diff
@@ -167,15 +228,19 @@ class ImagesDownloader {
     var downloadManager: DownloaderManager?
     
     init() {
-        onPublish = { imgName in
-            self.images.append(imgName)
-            //print("new img appended, img count: \(self.images.count)")
+        onPublish = { [weak self] imgName in
+            self?.images.append(imgName)
+            print("new img appended, img count: \(self!.images.count)")
```

Comment:
Why this unwrap is forced? 
sometimes it's good practice if you don't want to handle optional self you could try to unwrap it with guard, so after guard check we will be sure that self will be not optional
```suggestion
        onPublish = { [weak self] imgName in
            guard let self = self else {
                return
            }
            
            self.images.append(imgName)
            print("new img appended, img count: \(self.images.count)")
        }
```